pkg/codegen/go/gen_crd2pulumi.go

```diff
@@ -0,0 +1,69 @@
+package gen
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"path"
+
+	"github.com/pulumi/pulumi/pkg/v2/codegen/schema"
+)
+
+// GenCRDTypes returns a map from each module name to a buffer containing the
+// code for its generated types.
+func GenCRDTypes(tool string, pkg *schema.Package) (map[string]*bytes.Buffer, error) {
+	if err := pkg.ImportLanguages(map[string]schema.Language{"go": Importer}); err != nil {
+		return map[string]*bytes.Buffer{}, err
+	}
+
+	goInfo, _ := pkg.Language["go"].(GoPackageInfo)
+	packages := generatePackageContextMap(tool, pkg, goInfo)
+
+	var pkgMods []string
+	for mod := range packages {
+		pkgMods = append(pkgMods, mod)
+	}
+
+	buffers := map[string]*bytes.Buffer{}
+
+	for _, mod := range pkgMods {
+		pkg := packages[mod]
+		if len(pkg.types) > 0 {
+			buffer := &bytes.Buffer{}
+			pkg.genCRDHeader(buffer)
+			for _, t := range pkg.types {
+				pkg.genType(buffer, t)
+			}
+			pkg.genTypeRegistrations(buffer, pkg.types)
+			buffers[mod] = buffer
+		}
+	}
+
+	return buffers, nil
+}
+
+func (pkg *pkgContext) genCRDHeader(w io.Writer) {
+	fmt.Fprintf(w, "// *** This file was generated by crd2pulumi. ***\n\n")
+	var pkgName string
+	if pkg.mod == "" {
+		pkgName = pkg.pkg.Name
+	} else {
+		pkgName = path.Base(pkg.mod)
+	}
+	fmt.Fprintf(w, "package %s\n\n", pkgName)
+	fmt.Fprintf(w, imports)
+}
+
+const imports = `import (
+	"context"
+	"reflect"
+	"strings"
+
+	"github.com/fatih/structs"
```

Comment:
You probably won't need this. See my comment on the other PR: https://github.com/pulumi/pulumi-kubernetes/pull/1235#issuecomment-669997853